sympy/geometry/hyperbola.py

```diff
@@ -0,0 +1,124 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# In[ ]:
+
+
+import sympy
+import random
+import math
+import decimal
+import numpy as np
+import matplotlib.pyplot as plt
+from sympy import Symbol, symbols, plot_implicit
+from sympy import Rational
+from sympy.geometry import Point
+from sympy.core.symbol import _symbol
+
+
+class Hyperbola:
+    def __init__(self, hradius, vradius, angle=None, centre=None):
+        self.hradius = hradius
+        self.vradius = vradius
+        self.angle = 0
+        self._centre = Point(0,0)
+        if angle:
+            self.angle = angle
+        if centre:
+            self._centre = centre
+    
+    @property
+    def focus_distance(self):
+        return (sympy.sqrt(self.hradius ** 2 + self.vradius ** 2))
+    
+    @property
+    def foci(self):
+        return Point(self.focus_distance, 0), Point(- self.focus_distance, 0)
+        
+    @property
+    def eccentricity(self):
+        return self.focus_distance / self.hradius
+    
+    @property
+    def centre(self):
+        return self._centre
+    
+    def move(self, x1, y1):
+        self._centre = Point(x1, y1)
+    
+    def equation(self, x=None, y=None, expression=False):
+        if x:
```

Comment:
You probably want to check if x is a string here as it could make sense to allow it to be a Symbol.